nowcasting_dataset/manager.py

```diff
@@ -0,0 +1,384 @@
+                )
+                max_batch_id_on_disk = nd_fs_utils.get_maximum_batch_id(path)
+                first_batches_to_create[split_name][data_source_name] = max_batch_id_on_disk + 1
+
+        return first_batches_to_create
+
+    def _check_if_more_batches_are_required_for_split(
+        self,
+        split_name: split.SplitName,
+        first_batches_to_create: dict[split.SplitName, dict[str, int]],
+    ) -> bool:
+        """Returns True if batches still need to be created for any DataSource."""
+        n_batches_requested = self._get_n_batches_for_split_name(split_name.value)
+        for data_source_name in self.data_sources:
+            if first_batches_to_create[split_name][data_source_name] < n_batches_requested:
+                return True
+        return False
+
+    def _find_splits_which_need_more_batches(
+        self, first_batches_to_create: dict[split.SplitName, dict[str, int]]
+    ) -> list[split.SplitName]:
+        """Returns list of SplitNames which need more batches to be produced."""
+        splits_which_need_more_batches = []
+        for split_name in split.SplitName:
+            if self._check_if_more_batches_are_required_for_split(
+                split_name, first_batches_to_create
+            ):
+                splits_which_need_more_batches.append(split_name)
+        return splits_which_need_more_batches
+
+    def create_batches(self, overwrite_batches: bool) -> None:
+        """Create batches (if necessary).
+
+        Make dirs: `<output_data.filepath> / <split_name> / <data_source_name>`.
+
+        Also make `local_temp_path` if necessary.
+
+        Args:
+          overwrite_batches: If True then start from batch 0, regardless of which batches have
+            previously been written to disk. If False then check which batches have previously been
+            written to disk, and only create any batches which have not yet been written to disk.
+        """
+        first_batches_to_create = self._get_first_batches_to_create(overwrite_batches)
+
+        # Check if there's any work to do.
+        if overwrite_batches:
+            splits_which_need_more_batches = [split_name for split_name in split.SplitName]
+        else:
+            splits_which_need_more_batches = self._find_splits_which_need_more_batches(
+                first_batches_to_create
+            )
+            if len(splits_which_need_more_batches) == 0:
+                logger.info("All batches have already been created!  No work to do!")
+                return
+
+        # Load locations for each example off disk.
+        locations_for_each_example_of_each_split: dict[split.SplitName, pd.DataFrame] = {}
+        for split_name in splits_which_need_more_batches:
+            filename = self._filename_of_locations_csv_file(split_name.value)
+            logger.info(f"Loading {filename}.")
+            locations_for_each_example = pd.read_csv(filename)
```

Comment:
optional asserts that it has the correct columns?

Reply:
Good idea!  Done!